assignments/this.js

```diff
@@ -13,14 +13,58 @@
 
 // code example for Window Binding
 
+console.log("Global Binding \n ", this);
+
 // Principle 2
 
 // code example for Implicit Binding
 
+const profession = {
+    wish: "I want to be a",
+    time: "when I grow up!",
+    what: function (role) {
+        console.log("Implicit Binding \n ", `${this.wish} ${role} ${this.time}`);
+    }
+};
+
+profession.what("Wizard");
+
 // Principle 3
 
 // code example for New Binding
 
+function AnimalSounds (animal, sound){
+    this.animal = animal;
+    this.sound = sound;
+    this.speak = function (){
+        console.log("New Binding \n", ` A ${this.animal} makes the sound ${this.sound}`)
+    }
+};
+
+const dog = new AnimalSounds("dog", "Whoof Whoof!");
+const cat = new AnimalSounds("cat", "Meow");
+dog.speak();
+cat.speak();
+
 // Principle 4
 
-// code example for Explicit Binding
\ No newline at end of file
+// code example for Explicit Binding
+
```

Comment:
Can you think of a way to use a single console.log and achieve the same result? I'd check out *ternary operators* if you haven't heard of them.